tests/ui/nonminimal_bool_methods.rs

```diff
@@ -115,4 +115,66 @@ fn issue_12625() {
     if !(a as u64 <= b) {} //~ ERROR: this boolean expression can be simplified
 }
 
+fn issue_13436() {
+    fn not_zero(x: i32) -> bool {
+        x != 0
+    }
+
+    let opt = Some(500);
+    _ = opt.is_some_and(|x| x < 1000);
+    _ = opt.is_some_and(|x| x <= 1000);
+    _ = opt.is_some_and(|x| x > 1000);
+    _ = opt.is_some_and(|x| x >= 1000);
+    _ = opt.is_some_and(|x| x == 1000);
+    _ = opt.is_some_and(|x| x != 1000);
+    _ = opt.is_some_and(not_zero);
+    _ = !opt.is_some_and(|x| x < 1000); //~ ERROR: this boolean expression can be simplified
+    _ = !opt.is_some_and(|x| x <= 1000); //~ ERROR: this boolean expression can be simplified
+    _ = !opt.is_some_and(|x| x > 1000); //~ ERROR: this boolean expression can be simplified
+    _ = !opt.is_some_and(|x| x >= 1000); //~ ERROR: this boolean expression can be simplified
+    _ = !opt.is_some_and(|x| x == 1000); //~ ERROR: this boolean expression can be simplified
+    _ = !opt.is_some_and(|x| x != 1000); //~ ERROR: this boolean expression can be simplified
```

Comment:
Since `simplify_not` is called recursively on closures, could you add a recursive test? Something like this:

```rust
let opt_opt = Some(opt);
!opt_opt.is_some_and(|x| !x.is_some_and(|y| y != 1000));
```

I just want to make sure that the output is correct

Reply:
Nice catch! The test triggers an error `cannot replace slice of data that was already replaced`. I'm trying to fix it.

Reply:
Ahh, that happens if two suggestions overlap. Don't spend too much time on it. Overlapping suggestions are okay if they are still correct. You can also just create a second test file that doesn't run rustfix.

Reply:
Learned it! It looks like for users, rustfix will apply the first fix and ignore the subsequent. So `!opt_opt.is_some_and(|x| !x.is_some_and(|y| y != 1000))` will be fixed to `opt_opt.is_none_or(|x| x.is_some_and(|y| y != 1000))`.